pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -993,6 +992,11 @@ private void tryOverwriteOldProducer(Producer oldProducer, Producer newProducer)
                 handleProducerRemoved(oldProducer);
             }
         } else {
+            // If a producer with the same name tries to use a new connection, async check the old connection is
+            // available. The producers related the connection that not available are automatically cleaned up.
+            if (!Objects.equals(oldProducer.getCnx(), newProducer.getCnx())) {
+                oldProducer.getCnx().checkConnectionLiveness();
```

Comment:
This returns a future that can be used to prevent unnecessary exceptions. Did you consider wiring that up?

Reply:
@michaeljmarshall 

I submitted a new PR to improve this behavior. Please help to review it
- https://github.com/apache/pulsar/pull/21220